src/router/BaseRouter.php

```diff
@@ -27,22 +27,20 @@ abstract class BaseRouter<+TResponder> {
       $data = Dict\map($data, $value ==> \urldecode($value));
       return tuple($responder, new ImmMap($data));
     } catch (NotFoundException $e) {
-      foreach (HttpMethod::getValues() as $next) {
-        if ($next === $method) {
-          continue;
-        }
-        try {
-          list($responder, $data) = $resolver->resolve($next, $path);
-          if ($method === HttpMethod::HEAD && $next === HttpMethod::GET) {
-            $data = Dict\map($data, $value ==> \urldecode($value));
-            return tuple($responder, new ImmMap($data));
-          }
-          throw new MethodNotAllowedException();
-        } catch (NotFoundException $_) {
-          continue;
-        }
+      $allowed = $this->getAllowedMethods($path);
+      if (C\is_empty($allowed)) {
+        throw $e;
       }
-      throw $e;
+
+      if (
+        $method === HttpMethod::HEAD && $allowed === keyset[HttpMethod::GET]
```

Comment:
Should this be `C\contains_key($allowed, HttpMethod::GET)`?

e.g. if allowed is GET+POST then we still want to allow HEAD I think

Reply:
no, we shouldn't. that might cause a security issue, and the whole `HEAD <-> GET` rewrite should probably be removed in the future.

see : https://blog.teddykatz.com/2019/11/05/github-oauth-bypass.html

Reply:
Yep, this is an intentional behavior change that seemed reasonable to put into this diff at the same time